docs/DesignDocs/enclave_openssl_support.md

```diff
@@ -0,0 +1,90 @@
+# OE Enclave Runtime Support for OpenSSL
+
+## Motivation
+
+Per discussion in issue [#852](https://github.com/openenclave/openenclave/issues/852):
+
+- There are users of the OE SDK for which the existing mbedTLS library it ships with is insufficient to meet their needs.
+  - It is not a pervasively used crypto library, which increases the costs of porting existing code into the enclave.
```

Comment:
One question that comes up is why open enclave defaults to a `non pervasively used crypto library`?

Reply:
OE SDK used the mbedTLS library early on because:
- it has a smaller TCB than OpenSSL designed specifically for embedded systems with limited memory (as is the case with early SGX1 enclaves). It is designed to be statically linked and highly customizable to keep the TCB small by not including unused code, and had explicit configuration controls to limit its system dependencies on filesystem, sockets, memory allocation etc.
    - This is particularly relevant because OE only used the crypto library early with virtually no external dependencies by excluding the TLS library, whereas the equivalents in OpenSSL have a much larger dependency tree to port into the enclave runtime environment. Its use predates the introduction of the oesyscall compatibility layer, which the OpenSSL prototype support depends on heavily.
    - These are still relevant properties on the IoT side of the equation with the OP-TEE support, which is why the plan is not to deprecate support for mbedTLS. 
- at the point it was introduced, it also had a shorter and less frequent history of vulnerability disclosures.